src/python_tools/sensor_proto_log.py

```diff
@@ -0,0 +1,52 @@
+from .proto_log import ProtoLog
+from software.proto.sensor_msg_pb2 import SensorProto
+from google.protobuf.message import Message
+from typing import List, Dict
+
+
+class SensorProtoLog:
```

Comment:
It seems like `ProtoLog` does all the heavy lifting and we just use this as a wrapper around `ProtoLog` specialized to the `SensorProto`

Would I not be able to acheive the same thing  just doing
~~~
  sensor_proto_log = ProtoLog('/tmp/test/SensorProto', SensorProto)
  sensor_proto_log[1000].ssl_vision_msg
~~~
and not having a class like this altogether for every proto we want to analyze?

Reply:
because each `SensorMsg` has only one of `SSL_WrapperPacket`, `Referee`, or `RobotStatus`, it's less "ergonomic" to make the consumer of this API have to go `if sensor_proto_log[1000].HasField("ssl_vision_msg")` repeatedly rather than just making a class to 'demultiplex' these messages.